cmd/collector/app/grpc_handler.go

```diff
@@ -41,16 +40,24 @@ func NewGRPCHandler(logger *zap.Logger, spanProcessor SpanProcessor) *GRPCHandle
 
 // PostSpans implements gRPC CollectorService.
 func (g *GRPCHandler) PostSpans(ctx context.Context, r *api_v2.PostSpansRequest) (*api_v2.PostSpansResponse, error) {
-	// TODO
-	fmt.Printf("PostSpans(%+v)\n", *r)
-	for _, s := range r.Batch.Spans {
-		println(s.OperationName)
+	oks, err := g.spanProcessor.ProcessSpans(r.GetBatch().Spans, JaegerFormatType)
```

Comment:
I think this is loosing Batch.Process data.

Reply:
So the spans have pointers to the process. I'm not sure when that back-reference is set. If it is already set, there should not be a problem.

Reply:
by convention when a bunch of spans arrive in a Batch and logically they are all from the same Process then Span.Process will be nil, but Batch.Process will be defined. This is done to minimize the amount of data sent over the wire. So you need to loop through spans and copy the Process pointer from Batch if Span.Process == nil